### DISCUSSION TITLE
我这有一个问题希望请教一下

### DISCUSSION THREAD

**Discussion by bright5145** [PL-FE/bpmn-doc#5] [Q&A]
我在使用了自定义的palette后，如果再添加一个官方的工具栏module会出现异常。这种情况下我是需要也使用自定义的module实现嘛？

> this.bpmnModeler = new BpmnModeler({
                container: canvas,
                // entries改用js注入
                paletteEntries,
                // 添加控制板
                propertiesPanel: {
                    parent: '#js-properties-panel'
                },
                additionalModules: [
                    // 左边工具栏以及节点
                    // propertiesProviderModule,
                    // 自定义的palette
                    customPalette,
                    // 添加右边的工具栏后报错 No provider for "propertiesProvider"!
                    propertiesPanelModule,
                ],
                moddleExtensions: {
                  camunda: camundaModdleDescriptor
                }
        })

**Comment by PL-FE** [comment]
需要引入配置。
```js
 // 右侧属性栏
import propertiesPanelModule from 'bpmn-js-properties-panel'
import propertiesProviderModule from 'bpmn-js-properties-panel/lib/provider/camunda'
 // 一个描述的json
import camundaModdleDescriptor from 'camunda-bpmn-moddle/resources/camunda'

const bpmnModeler = new BpmnModeler({
	//添加控制板
  propertiesPanel: {
  	parent: '#js-properties-panel'
  },
  additionalModules: [
  	propertiesPanelModule,
  	propertiesProviderModule
  ],
  moddleExtensions: {
    //如果要在属性面板中维护camunda：XXX属性，则需要此 
    camunda: camundaModdleDescriptor
  }
})
```

友情链接一个大佬，有详细介绍[LinDaiDai](https://github.com/LinDaiDai/bpmn-chinese-document/blob/master/LinDaiDai/%E5%85%A8%E7%BD%91%E6%9C%80%E8%AF%A6bpmn.js%E6%95%99%E6%9D%90-properties-panel%E7%AF%87(%E4%B8%8A).md)
再者，可以阅读下这个demo [bpmn-camunda](https://github.com/PL-FE/bpmn-camunda)，也实现了`propertiesPanelModule `
